stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/ChannelRepository.kt

```diff
@@ -1,85 +1,73 @@
-import io.getstream.chat.android.livedata.entity.ChannelEntity
-import io.getstream.chat.android.livedata.extensions.isPermanent
 import io.getstream.chat.android.livedata.repository.mapper.toEntity
+import io.getstream.chat.android.livedata.repository.mapper.toModel
+import java.util.Date
 
 internal class ChannelRepository(
-    var channelDao: ChannelDao,
-    var cacheSize: Int = 100,
-    var currentUser: User,
-    var client: ChatClient
+    private val channelDao: ChannelDao,
+    cacheSize: Int = 100,
 ) {
-    // the channel cache is simple, just keeps the last 100 users in memory
-    var channelCache = LruCache<String, ChannelEntity>(cacheSize)
+    // the channel cache is simple, just keeps the last several users in memory
+    private val channelCache = LruCache<String, Channel>(cacheSize)
 
-    suspend fun insert(channelEntity: ChannelEntity) {
-        insert(listOf(channelEntity))
+    suspend fun insert(channel: Channel) {
+        updateCache(listOf(channel))
+        channelDao.insert(channel.toEntity())
     }
 
-    private fun updateCache(channelEntities: List<ChannelEntity>) {
-        for (channelEntity in channelEntities) {
-            channelCache.put(channelEntity.cid, channelEntity)
+    private fun updateCache(channels: Collection<Channel>) {
+        for (channel in channels) {
+            channelCache.put(channel.cid, channel)
         }
     }
 
     suspend fun insertChannels(channels: Collection<Channel>) {
-        insert(channels.map(Channel::toEntity))
-    }
-
-    private suspend fun insert(channelEntities: List<ChannelEntity>) {
-        if (channelEntities.isEmpty()) return
-        channelDao.insertMany(channelEntities)
-        updateCache(channelEntities)
+        if (channels.isEmpty()) return
+        updateCache(channels)
+        channelDao.insertMany(channels.map(Channel::toEntity))
     }
 
     suspend fun delete(cid: String) {
         channelCache.remove(cid)
         channelDao.delete(cid)
     }
 
-    suspend fun select(cid: String): ChannelEntity? {
-        return select(listOf(cid)).getOrElse(0) { null }
+    suspend fun select(
+        cid: String,
+        getUser: suspend (userId: String) -> User,
+        getMessage: suspend (messageId: String) -> Message?,
+    ): Channel? {
+        return select(listOf(cid), getUser, getMessage).getOrNull(0)
     }
 
-    suspend fun select(channelCIDs: List<String>): List<ChannelEntity> {
-        val cachedChannels: MutableList<ChannelEntity> = mutableListOf()
-        for (cid in channelCIDs) {
-            val channelEntity = channelCache.get(cid)
-            channelEntity?.let { cachedChannels.add(it) }
-        }
+    suspend fun select(
+        channelCIDs: List<String>,
+        getUser: suspend (userId: String) -> User,
+        getMessage: suspend (messageId: String) -> Message?,
+    ): List<Channel> {
+        val cachedChannels: MutableList<Channel> = channelCIDs.mapNotNullTo(mutableListOf()) { channelCache.get(it) }
```

Comment:
You can use just `channelCache::get` here

Reply:
Thanks! Nice, love fun references, seems more elegant for me